addons/website_hr_recruitment/models/hr_job.py

```diff
@@ -132,4 +135,15 @@ def _search_get_detail(self, website, order, options):
             'fetch_fields': fetch_fields,
             'mapping': mapping,
             'icon': 'fa-briefcase',
+            'template_key' : 'website_hr_recruitment.search_items_jobs',
+            'group_title' : self.env._('Jobs'),
         }
+
+    def _search_render_results(self, fetch_fields, mapping, icon, limit):
+        results_data = super()._search_render_results(fetch_fields, mapping, icon, limit)
+        for data in results_data:
+            job = self.browse(data['id'])
+            data['department_name'] = job.department_id.name if job.department_id else ''
+            # res.partner address_id is not available to public - sudo()
+            data['address'] = job.sudo().address_id.name if job.address_id else ''
```

Comment:
```suggestion
            data['address'] = job.address_id.name or ''
```
Why not sudo job object itself as we are access department_id and address_id for read purpose only